### PR TITLE
[SAPBTPCFS-3288] CUSTOM TEMPLATES FOR SECRETS

### PR DESCRIPTION
**Pull request by I065450**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -319,77 +322,204 @@ spec:
...
+  plan: sample-plan // The service plan name                
+  type: sample-service  // The service offering name
 ```
 
 ###### Credentials as JSON Object
-To show credentials returned from the broker as a JSON object, use the 'secretKey' attribute in the service binding spec.
 
-The value of 'secretKey' is the name of the key that stores the credentials in JSON format.
+To show credentials returned from the broker within the `Secret` resource  as a JSON object, use the 'secretKey' attribute in the `ServiceBinding` spec.
+The value of this 'secretKey' must be the name of the key that stores the credentials in JSON format:
```
Comment (avilupu) [inline README.md:378]: not "must be"

e.g., The value of 'secretKey' would the name of a key which stores the credentials in JSON format

File: README.md [hunk 2]
```diff
@@ -319,77 +322,204 @@ spec:
...
-plan: sample-plan // The service plan name
-type: sample-service // The service offering name
+```yaml
+apiVersion: services.cloud.sap.com/v1
+kind: ServiceBinding
+metadata:
+  name: sample-binding
+spec:
+  serviceInstanceName: sample-instance
+  secretKey: your-secretKey-value
```
Comment (avilupu) [inline README.md:389]: use 'myCredentialData'

File: README.md [hunk 3]
```diff
@@ -319,77 +322,204 @@ spec:
...
+Input:
+
+```yaml
+apiVersion: services.cloud.sap.com/v1
+kind: ServiceBinding
+metadata:
+  name: sample-binding
+spec:
+  serviceInstanceName: sample-instance
+  secretRootKey: your-secretRootKey-value
```
Comment (avilupu) [inline README.md:426]: use 'myBindingData'

File: README.md [hunk 4]
```diff
@@ -319,77 +322,204 @@ spec:
...
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      labels:
+        service_plan: {{ .instance.plan }}
+      annotations:
+        instance: {{ .instance.instance_name }}
+    stringData:
+      username: {{ .credentials.client_id }}
+      password: {{ .credentials.client_secret }}
```
Comment (avilupu) [inline README.md:488]: change to capital letters
USERNAME
PASSWORD

File: README.md [hunk 5]
```diff
@@ -319,77 +322,204 @@ spec:
...
+
+```yaml
+apiVersion: services.cloud.sap.com/v1
+kind: ServiceBinding
+metadata:
+  name: sample-binding
+spec:
+  serviceInstanceName: sample-instance
+  secretTemplate: your-Go-template
+```
```
Comment (avilupu) [inline README.md:468]: what is "your-Go-template"?

File: README.md [hunk 6]
```diff
@@ -319,77 +322,204 @@ spec:
...
+      labels:
+        service_plan: {{ .instance.plan }}
+      annotations:
+        instance: {{ .instance.instance_name }}
+    stringData:
+      username: {{ .credentials.client_id }}
+      password: {{ .credentials.client_secret }}
+```
+The sample-binding map on which your-Go-template operates is in the JSON format:
+
```
Comment (avilupu) [inline README.md:491]: not sure I understand...

File: README.md [hunk 7]
```diff
@@ -572,20 +702,21 @@ secret-parameter:
...
+| credentialsRotationPolicy.rotationFrequency | `duration`  | Specifies the frequency at which the binding rotation is performed.                                                                                                                                                                                                                                                                                                 |
+| credentialsRotationPolicy.rotatedBindingTTL | `duration`  | Specifies the time period for which to keep the rotated binding.                                                                                                                                                                                                                                                                                                    |
+| SecretTemplate | `string`  | A Go template is used to generate a custom Kubernetes v1/Secret, using data from both the service binding returned by Service Manager and instance information.
```
Comment (avilupu) [inline README.md:719]: add link

**Review by avilupu — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -319,77 +322,204 @@ spec:
     
     ```
 
-4.  Check that a secret with the same name as the name of your binding is created. The secret contains the service credentials that apps in your cluster can use to access the service.
+3.  Check that the `Secret` with the same name as the name of your `ServiceBinding` is created. Remember, you need this custom resource to obtain the access credentials needed for the apps to use the service:
```
[the remark this answers is not in the document]
Comment (danielforsap) [reply README.md:325]: Done

File: README.md [hunk 9]
```diff
@@ -319,77 +322,204 @@ spec:
...
+Even though `Secret` resources can come in various formats, they all share a common basic content. The parameters within the `Secret` fall into two categories:
+
+- Credentials returned from the broker: These credentials allow your applications to connect to the `ServiceInstance`.
+- Attributes of the associated `ServiceInstance`: This information provides details about the service instance itself.
+  
+Now let's explore these various formats:
+
+###### Key-Value Pairs (Default)
+
+Input:
```
[the remark this answers is not in the document]
Comment (danielforsap) [reply README.md:350]: Done

**Review by danielforsap — CHANGES_REQUESTED** [review]
3 issues still open.

File: README.md (hunk 3 above)
Comment (danielforsap) [reply README.md:428]: Wouldn't this be myCredentialsAndInstanceData if already?? @avilupu

File: README.md (hunk 6 above)
Comment (danielforsap) [reply README.md:493]: @avilupu  I think we are still missing info here. Go template is applied to a data structure to generate an output. In this case, the generated output is the Secret format that we want. However, there is also a map. We need to explain more where this map comes from. @I065450

**Review by danielforsap — APPROVED** [review]

**Pull request merged**